libs/dal/src/repositories/notification-template/notification-template.repository.ts

```diff
@@ -46,22 +46,56 @@ export class NotificationTemplateRepository extends BaseRepository<
     const requestQuery: NotificationTemplateQuery = {
       _id: id,
       isBlueprint: true,
-      _organizationId: NotificationTemplateRepository.getBlueprintOrganizationId() as string,
+      _environmentId: NotificationTemplateRepository.getBlueprintEnvironmentId() as string,
     };
 
     const item = await this.MongooseModel.findOne(requestQuery).populate('steps.template');
 
     return this.mapEntity(item);
   }
 
+  async findAllGroupedByCategory(): Promise<{ name: string; blueprints: NotificationTemplateEntity[] }[]> {
+    if (!NotificationTemplateRepository.getBlueprintEnvironmentId()) {
+      return {} as { name: string; blueprints: NotificationTemplateEntity[] }[];
+    }
+
+    const requestQuery: NotificationTemplateQuery = {
+      isBlueprint: true,
+      _environmentId: NotificationTemplateRepository.getBlueprintEnvironmentId() as string,
+    };
+
+    const items = await this.MongooseModel.find(requestQuery)
+      .populate('steps.template')
+      .populate('notificationGroup')
+      .lean();
```

Comment:
If here we won't have duplicates I think we could simplify and just use a map function rather than a reduce when calculated the groupedItems.

Reply:
not sure if we can do that, example plz?

Reply:
> If here we won't have duplicates I think we could simplify and just use a map function rather than a reduce when calculated the groupedItems.

Not sure if I got the 'map' right in your suggestion.

Reply:
> not sure if we can do that, example plz?

Was this comment directed to me or Pablo? :)

Reply:
to @p-fernandez

Reply:
I just noticed that there could be multiple blueprints for different notification groups. I missed that point, sorry. 😓